blob/blob.go

```diff
@@ -326,6 +356,14 @@ type WriterOptions struct {
 	// Keys may not be empty, and are lowercased before being written.
 	// Duplicate case-insensitive keys (e.g., "foo" and "FOO") are an error.
 	Metadata map[string]string
+
+	// Callback is a callback that will be called at most once, before any data
+	// is written. asFunc can be used as an escape hatch for accessing
```

Comment:
Will it be called on a zero-length blob? Now that I'm thinking about it, it probably should always be called, but the guarantee is that it will be called before `Close`, but not necessarily before the first user-facing `Write`.

Reply:
I updated the docstring a bit to say it will always be called exactly once, unless NewWriter or the equivalent returns an error.

I think it has to be called before the first actual write (not the first time the user calls Write); also clarified.